src/asio.cpp

```diff
@@ -91,63 +93,91 @@ namespace fc {
         }
     }
 
-    struct default_io_service_scope
+    static int16_t default_num_io_threads = 0;
```

Comment:
I don't like the semantics of that variable, it mixes up too many things:
* it is the default setting for the number of threads...
* ...until default_io_service_scope is instanciated, then it becomes the number of active threads...
* ...until set_default_num_threads is called (again?), then it becomes whatever was passed to the call.

Reply:
How about ``almost_default_num_io_threads_until_changed_then_probable_num_threads``?

Just kidding. What about ``num_io_threads``? I thought about ways to prevent calling ``set_`` (whose name will change too), but it would require another static to indicate if the constructor has been called or not. Yuck, but do-able if you think it is important. Instead, I had the ``set_`` return the old value, which moves the responsibility of seeing if the call will actually do something to the caller.

Reply:
How about throwing in set_ if the current value is != 0, and making the actual number an instance member? (Not sure what's best either.)

Reply:
Done